src/middlewared/middlewared/alert/source/volume_status.py

```diff
@@ -1,62 +1,70 @@
 from middlewared.alert.base import AlertClass, AlertCategory, AlertLevel, Alert, AlertSource
-from middlewared.alert.schedule import CrontabSchedule
 
 
 class VolumeStatusAlertClass(AlertClass):
     category = AlertCategory.STORAGE
     level = AlertLevel.CRITICAL
-    title = "Pool Status Is Not Healthy"
-    text = "Pool %(volume)s state is %(state)s: %(status)s%(devices)s"
+    title = 'Pool Status Is Not Healthy'
+    text = 'Pool %(volume)s state is %(state)s: %(status)s%(devices)s'
+    proactive_support = True
+
 
+class BootPoolStatusAlertClass(AlertClass):
+    category = AlertCategory.SYSTEM
+    level = AlertLevel.CRITICAL
+    title = 'Boot Pool Is Not Healthy'
+    text = 'Boot pool status is %(status)s: %(status_detail)s.'
     proactive_support = True
 
 
 class VolumeStatusAlertSource(AlertSource):
+    async def check(self):
+        alerts = []
+        boot_name = await self.middleware.call('boot.pool_name')
+        for guid, info in (await self.middleware.call('zfs.pool.query_imported_fast')).items():
```

Comment:
This is consuming the process pool ?

Reply:
The function is actually simple enough that we could probably move it to the ZFS utils file and import directly into the volume status alert.

Reply:
sorry, I didn't see these but that's exactly what I'm doing.